src/js/actions/layereffects.js

```diff
@@ -179,6 +179,39 @@ define(function (require, exports) {
     setShadowEnabled.writes = [locks.PS_DOC, locks.JS_DOC];
 
     /**
+     * Delete the selected Shadow for all selected layers
+     *
+     * @param {Document} document
+     * @param {Immutable.Iterable.<Layer>} layers list of layers to update
+     * @param {number} shadowIndex index of the Drop Shadow within the layer(s)
```

Comment:
I see you `forEach` to calculate actual index list. Please refer to bug I added as a comment.